plugins/events-backend-module-http/src/service/HttpEndpointEventPublisher.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorHandler } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { EventPublisher, EventBus } from '@backstage/plugin-events-node';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { EventHttpRequestValidator } from './types';
+
+/**
+ * Publishes events received from their origin (e.g., webhook events from an SCM system)
+ * via HTTP endpoint and passes the request body as event payload to the registered subscribers.
+ *
+ * @public
+ */
+export class HttpEndpointEventPublisher implements EventPublisher {
+  private eventBus?: EventBus;
+
+  static fromConfig(env: {
+    config: Config;
+    logger: Logger;
+    router: express.Router;
+    validators?: Record<string, EventHttpRequestValidator>;
+  }): HttpEndpointEventPublisher {
+    const topics = env.config.getStringArray('events.modules.http.topics');
+    return new HttpEndpointEventPublisher(
+      env.logger,
+      env.router,
+      topics,
+      env.validators,
+    );
+  }
+
+  private constructor(
+    private logger: Logger,
+    router: express.Router,
+    topics: string[],
+    validators?: Record<string, EventHttpRequestValidator>,
+  ) {
+    router.use(this.createRouter(topics, validators));
+  }
+
+  async connect(eventBus: EventBus): Promise<void> {
+    this.eventBus = eventBus;
+  }
+
+  private createRouter(
+    topics: string[],
+    validators?: Record<string, EventHttpRequestValidator>,
+  ): express.Router {
+    const router = Router();
+    router.use(express.json());
+    router.use(errorHandler());
```

Comment:
i _think_ you want this one at the bottom, just before returning. That's what we do elsewhere at least 😅

Reply:
Does it make a difference? 🤔 It just adds the ErrorRequestHandler and afterwards, only post request handler will be added. Therefore, I assume it doesn't really matter. However, to keep consistency in the codebase, I will change it.